### PR TITLE
Make MDS tests skippable if MDS is not installed

### PR DESCRIPTION
**Pull request by danielballan**

### CODE REVIEW

File: bluesky/tests/test_vertical_integration.py [hunk 1]
```diff
@@ -1,17 +1,33 @@
...
 
 def test_scan_and_get_data():
+    try:
+        import metadatastore
+    except ImportError:
+        raise SkipTest
+    try:
+        from dataportal import DataBroker as db
+    except ImportError:
+        raise SkipTest
```
Comment (ericdill) [inline bluesky/tests/test_vertical_integration.py]: Why not combine these two try/except blocks?

File: bluesky/tests/test_vertical_integration.py [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+from nose import SkipTest
 from bluesky.examples import stepscan, det, motor
-from bluesky.standard_config import gs
 
 def setup():
-    mds_setup()
+    try:
+        from metadatastore.utils.testing import mds_setup
+    except ImportError:
+        pass  # test will be skipped
```
Comment (ericdill) [inline bluesky/tests/test_vertical_integration.py]: in-line comment is incorrect

File: bluesky/hardware_checklist.py [hunk 3]
```diff
@@ -17,10 +16,12 @@
 
 
 def connect_mds_mongodb():
+    import metadatastore.conf
     _connect_mongodb('metadatastore', metadatastore.conf.connection_config)
 
 
 def connect_fs_mongodb():
+    import metadatastore.conf
```
Comment (ericdill) [inline bluesky/hardware_checklist.py]: this should be `import filestore.conf`

File: bluesky/hardware_checklist.py [hunk 4]
```diff
@@ -5,7 +5,6 @@
 import logging
 from collections import namedtuple
 import requests
-import metadatastore.conf
 import filestore.conf
```
Comment (ericdill) [inline bluesky/hardware_checklist.py]: I imagine you want to remove this line too

File: bluesky/tests/test_vertical_integration.py (hunk 2 above)
Comment (danielballan) [inline bluesky/tests/test_vertical_integration.py]: How?

File: bluesky/tests/test_vertical_integration.py (hunk 1 above)
Comment (danielballan) [inline bluesky/tests/test_vertical_integration.py]: could make more informative errors later, individualized

**Comment by ericdill** [comment]
Do you want to follow the same pattern and make pyolog optional in hardware_checklist too?

File: bluesky/tests/test_vertical_integration.py (hunk 1 above)
Comment (ericdill) [inline bluesky/tests/test_vertical_integration.py]: ok

File: bluesky/tests/test_vertical_integration.py (hunk 2 above)
Comment (ericdill) [inline bluesky/tests/test_vertical_integration.py]: I guess it is not incorrect if enough context is placed around the in-line comment. I initially assumed that this was just a copy/paste error from the `test_scan_and_get_data` function

**Comment by danielballan** [comment]
Good point about pyolog. Added a commit.

**Comment by ericdill** [comment]
looks good to me

**Comment by ericdill** [comment]
tests are passing locally?

**Pull request merged**